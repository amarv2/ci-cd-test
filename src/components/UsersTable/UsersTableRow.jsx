import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { connect } from 'react-redux';
import history from '../../utils/History/history';

import styles from './UsersTable.module.scss';

class UsersTableRow extends Component {
    editUserById = () => {
      this.props.getUserById(this.props.urlParams, this.props.data.id);
      history.push(`/organizations/${this.props.urlParams}/users/${this.props.data.id}/edit`);
      this.props.toggleEditMode();
    }

    toggleConfirmationModal = (category) => {
      this.props.getUserById(this.props.urlParams, this.props.data.id);
      this.props.toggleConfirmationModal(category);
    }

    goToUserDetails = () => {
      this.props.getUserById(this.props.urlParams, this.props.data.id);
      history.push(`/organizations/${this.props.urlParams}/users/${this.props.data.id}`);
    }

    render() {
      return (
        <tr className={`${this.props.data.active ? '' : styles.blockedUser}`}>
          <td>{ this.props.data.name }</td>
          <td>{ this.props.data.surname }</td>
          <td>{ this.props.data.email }</td>
          <td>{ this.props.data.githubHandle }</td>
          <td className="text-center cursor-pointer">
            <div
              onClick={() => { this.editUserById(); }}
              role="presentation"
            >
              <i className="far fa-edit" />
            </div>
          </td>
          <td className="text-center cursor-pointer">
            {this.props.data.active ? (
              <div
                onClick={() => { this.toggleConfirmationModal('blockUser'); }}
                role="presentation"
              >
                <i className="fas fa-ban" />
              </div>
            ) : (
              <div
                onClick={() => { this.toggleConfirmationModal('unblockUser'); }}
                role="presentation"
              >
                <i className="fas fa-check-circle" />
              </div>
            )}
          </td>
          <td className="text-center cursor-pointer">
            <div
              onClick={() => { this.toggleConfirmationModal('deleteUser'); }}
              role="presentation"
            >
              <i className="far fa-trash-alt" />
            </div>
          </td>
          <td className="text-center cursor-pointer">
            <div
              onClick={() => { this.goToUserDetails(); }}
              role="presentation"
            >
              <i className="far fa-eye" />
            </div>
          </td>
        </tr>
      );
    }
}

function mapDispatchToProps(dispatch) {
  return {
    toggleConfirmationModal: (modalCategory) => {
      dispatch({ type: 'userConstants.TOGGLE_CONFIRMATION_MODAL', modalCategory });
    },
    getUserById: (organizationName, userId) => {
      dispatch({ type: 'userConstants.GET_USER_BY_ID_REQUEST', payload: { organizationName, userId } });
    },
    toggleEditMode: () => {
      dispatch({ type: 'userConstants.USER_EDIT_MODE_TOGGLE' });
    },
  };
}

UsersTableRow.propTypes = {
  data: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    surname: PropTypes.string.isRequired,
    email: PropTypes.string.isRequired,
    githubHandle: PropTypes.string.isRequired,
    active: PropTypes.bool.isRequired,
  }).isRequired,
  urlParams: PropTypes.string.isRequired,
  toggleEditMode: PropTypes.func.isRequired,
  getUserById: PropTypes.func.isRequired,
  toggleConfirmationModal: PropTypes.func.isRequired,
};

export default connect(null, mapDispatchToProps)(UsersTableRow);
