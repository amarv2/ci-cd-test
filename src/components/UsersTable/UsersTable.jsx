import React from 'react';
import PropTypes from 'prop-types';

import UsersTableRow from './UsersTableRow';

const UsersTable = ({ data, urlParams }) => (
  <table className="table table-hover">
    <thead>
      <tr>
        <th>First Name</th>
        <th>Last Name</th>
        <th>Email</th>
        <th>Github Handle</th>
        <th className="text-centered">Edit</th>
        <th className="text-centered">Block</th>
        <th className="text-centered">Delete</th>
        <th className="text-centered">View Details</th>
      </tr>
    </thead>
    <tbody>
      { data
        && data.users.map((users) => (
          <UsersTableRow key={users.id} data={users} urlParams={urlParams} />
        ))
      }
    </tbody>
  </table>
);

UsersTable.propTypes = {
  data: PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    surname: PropTypes.string,
    githubHandle: PropTypes.string,
  }).isRequired,
  urlParams: PropTypes.string.isRequired,
};

export default UsersTable;
