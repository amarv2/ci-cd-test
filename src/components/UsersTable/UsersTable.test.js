/* global it, expect, describe */
import React from 'react';
import { shallow } from 'enzyme';

import UsersTable from './UsersTable';

describe('UserTable', () => {
  it('snapshot for userTable', () => {
    const props = {
      data: {
        users: [],
      },
      urlParams: '',
    };
    const tree = shallow(<UsersTable {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
