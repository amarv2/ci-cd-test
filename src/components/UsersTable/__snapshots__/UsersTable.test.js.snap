// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserTable snapshot for userTable 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UsersTable
    data={
      Object {
        "users": Array [],
      }
    }
    urlParams=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <thead>
          <tr>
            <th>
              First Name
            </th>
            <th>
              Last Name
            </th>
            <th>
              Email
            </th>
            <th>
              Github Handle
            </th>
            <th
              className="text-centered"
            >
              Edit
            </th>
            <th
              className="text-centered"
            >
              Block
            </th>
            <th
              className="text-centered"
            >
              Delete
            </th>
            <th
              className="text-centered"
            >
              View Details
            </th>
          </tr>
        </thead>,
        <tbody />,
      ],
      "className": "table table-hover",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <tr>
            <th>
              First Name
            </th>
            <th>
              Last Name
            </th>
            <th>
              Email
            </th>
            <th>
              Github Handle
            </th>
            <th
              className="text-centered"
            >
              Edit
            </th>
            <th
              className="text-centered"
            >
              Block
            </th>
            <th
              className="text-centered"
            >
              Delete
            </th>
            <th
              className="text-centered"
            >
              View Details
            </th>
          </tr>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <th>
                First Name
              </th>,
              <th>
                Last Name
              </th>,
              <th>
                Email
              </th>,
              <th>
                Github Handle
              </th>,
              <th
                className="text-centered"
              >
                Edit
              </th>,
              <th
                className="text-centered"
              >
                Block
              </th>,
              <th
                className="text-centered"
              >
                Delete
              </th>,
              <th
                className="text-centered"
              >
                View Details
              </th>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "First Name",
              },
              "ref": null,
              "rendered": "First Name",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Last Name",
              },
              "ref": null,
              "rendered": "Last Name",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Email",
              },
              "ref": null,
              "rendered": "Email",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Github Handle",
              },
              "ref": null,
              "rendered": "Github Handle",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Edit",
                "className": "text-centered",
              },
              "ref": null,
              "rendered": "Edit",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Block",
                "className": "text-centered",
              },
              "ref": null,
              "rendered": "Block",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Delete",
                "className": "text-centered",
              },
              "ref": null,
              "rendered": "Delete",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "View Details",
                "className": "text-centered",
              },
              "ref": null,
              "rendered": "View Details",
              "type": "th",
            },
          ],
          "type": "tr",
        },
        "type": "thead",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
        },
        "ref": null,
        "rendered": Array [],
        "type": "tbody",
      },
    ],
    "type": "table",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <thead>
            <tr>
              <th>
                First Name
              </th>
              <th>
                Last Name
              </th>
              <th>
                Email
              </th>
              <th>
                Github Handle
              </th>
              <th
                className="text-centered"
              >
                Edit
              </th>
              <th
                className="text-centered"
              >
                Block
              </th>
              <th
                className="text-centered"
              >
                Delete
              </th>
              <th
                className="text-centered"
              >
                View Details
              </th>
            </tr>
          </thead>,
          <tbody />,
        ],
        "className": "table table-hover",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <tr>
              <th>
                First Name
              </th>
              <th>
                Last Name
              </th>
              <th>
                Email
              </th>
              <th>
                Github Handle
              </th>
              <th
                className="text-centered"
              >
                Edit
              </th>
              <th
                className="text-centered"
              >
                Block
              </th>
              <th
                className="text-centered"
              >
                Delete
              </th>
              <th
                className="text-centered"
              >
                View Details
              </th>
            </tr>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <th>
                  First Name
                </th>,
                <th>
                  Last Name
                </th>,
                <th>
                  Email
                </th>,
                <th>
                  Github Handle
                </th>,
                <th
                  className="text-centered"
                >
                  Edit
                </th>,
                <th
                  className="text-centered"
                >
                  Block
                </th>,
                <th
                  className="text-centered"
                >
                  Delete
                </th>,
                <th
                  className="text-centered"
                >
                  View Details
                </th>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "First Name",
                },
                "ref": null,
                "rendered": "First Name",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Last Name",
                },
                "ref": null,
                "rendered": "Last Name",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Email",
                },
                "ref": null,
                "rendered": "Email",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Github Handle",
                },
                "ref": null,
                "rendered": "Github Handle",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Edit",
                  "className": "text-centered",
                },
                "ref": null,
                "rendered": "Edit",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Block",
                  "className": "text-centered",
                },
                "ref": null,
                "rendered": "Block",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Delete",
                  "className": "text-centered",
                },
                "ref": null,
                "rendered": "Delete",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "View Details",
                  "className": "text-centered",
                },
                "ref": null,
                "rendered": "View Details",
                "type": "th",
              },
            ],
            "type": "tr",
          },
          "type": "thead",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
          },
          "ref": null,
          "rendered": Array [],
          "type": "tbody",
        },
      ],
      "type": "table",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
